code/extension/Member.php

```diff
@@ -26,6 +26,8 @@ class Member extends \DataExtension
 
     private static $admins_can_disable = false;
 
+    private static $regenerate_on_activation = true;
```

Comment:
Current behaviour/non-breaking = correct
But the regeneration upon activation also is current behaviour/non-breaking. Setting regenerate_on_activation to false is a new configuration option, which changes the behaviour from re-generating a new secret upon each new activation (current/default behaviour), to generating a secret only once and NOT auto-regenerating it between activations.